client/state/happychat/middleware.js

```diff
@@ -193,6 +262,27 @@ export const sendAnalyticsLogEvent = ( connection, { getState }, { meta: { analy
 	} );
 };
 
+export const sendActionLogsAndEvents = ( connection, { getState }, action ) => {
+	const state = getState();
+
+	// If there's not an active Happychat session, do nothing
+	if ( ! isHappychatClientConnected( state ) || ! isHappychatChatAssigned( state ) ) {
+		return false;
+	}
+
+	// If there's analytics metadata attached to this action, send analytics events
+	if ( has( action, 'meta.analytics' ) ) {
+		sendAnalyticsLogEvent( connection, action );
```

Comment:
Since we don't early return here, actions with analytics data will flow through the rest of the function. Are we intentionally doing this? If not, we might want to early return here so that whoever touching analytics actions in the future can have less mental burden :)